tools/socket_passing.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+# This tool is a helper script that queries the admin address for all listener
+# addresses after envoy startup. (The admin adress is written out to a file by
+# setting the -a flag in the envoy binary.) The script then outputs a new json
+# config file with updated listener addresses. This script is currently called
+# in the hot restart integration test to update listener addresses bound to
+# port 0 in the intial json config file.
+
+from collections import OrderedDict
+
+import argparse
+import httplib
+import json
+import os.path
+import sys
+import time
+
+# Seconds to wait for the admin address output file to appear. The script exits
+# if the file is not found.
+ADMIN_FILE_TIMEOUT_SECS = 20
+
+def ReplaceListenerAddresses(original_json, admin_address, updated_json):
```

Comment:
I think this name is now legacy, it should probably be `PrintListenerAddresses` or something you think is better.